include/OpenSpaceToolkit/Mathematics/CurveFitting/Interpolator.hpp

```diff
@@ -16,46 +18,60 @@ namespace curvefitting
 
 using ostk::core::type::Real;
 using ostk::core::type::Size;
+using ostk::core::type::Shared;
 
 using ostk::mathematics::object::VectorXd;
 
-/// @brief                      Interpolator
+/// @brief Interpolator (abstract class)
 ///
-///                             In mathematics, an interpolator is a type of estimator allowing to construct new data
-///                             points based on a range of a discrete set of known data points.
+/// In mathematics, an interpolator is a type of estimator allowing to construct new data
+/// points based on a range of a discrete set of known data points.
 ///
-/// @ref                        https://en.wikipedia.org/wiki/Interpolator.
-
+/// @ref https://en.wikipedia.org/wiki/Interpolator.
 class Interpolator
 {
    public:
-    /// @brief              Default constructor
-    ///
-    Interpolator() = default;
+    enum class InterpolationType
+    {
+        BarycentricRational,
+        CubicSpline,
+        Linear
+    };
 
-    /// @brief              Destructor (pure virtual)
+    /// @brief Constructor
+    Interpolator(const InterpolationType& anInterpolationType);
```

Comment:
```suggestion
    Interpolator(const InterpolationType& anInterpolationType, const VectorXd& anXVector, const VectorXd& aYVector);
```
Since these are used by every interpolator, we could make it a part of the parent class.
You can add getters as well.

Reply:
I see what you mean, the drawback here is that we increase the coupling between all the interpolators and the base class

Reply:
Yeah that's a fair point about double storing the data.